developing_SoCCo.py

```diff
@@ -0,0 +1,471 @@
+    pbc=sc.perceivedBehavioralControlF(10)
+    psn=sc.perceivedSocialNorm(emissionsPC)
+    emissionsPC_Del = sc.eIncrement(att, pbc, psn)
+    emissionsPC_New = emissionsPC_Del + emissionsPC
+    sc.updatePCEmissions(emissionsPC_New, eff, pbc, testClimateData, 3,riskSens=1.0)
+
+
+
+if False: ### WORKS!
+    co2_ts=np.linspace(290, 300, 4)
+    popTotal=7130010000 # Wolfram: QuantityMagnitude[CountryData["World", "Population"]]
+    popN=sc.popIntoNgroups(popTotal,nGroups=10)
+    pcE_ts=sc.randomNormalF(5.049, 0.5, 10)
+    pcE_ts=np.atleast_2d(pcE_ts).transpose()
+    tData_ts=np.array([0,0.1,0.2,0.1]) # temperature
+    
+    eff=sc.efficacyF(10)
+    pbc=sc.perceivedBehavioralControlF(10)
+    # yearCurrent=0
+    percepWindowSize=3
+    riskSens=1.0
+    
+    pcE_ts,tData_ts,co2_ts = sc.iterateOneStep(pcE_ts,tData_ts, co2_ts, eff, pbc,popN,
+        percepWindowSize=3,riskSens=1.0)
+        
+    pcE_ts,tData_ts,co2_ts = sc.iterateOneStep(pcE_ts,tData_ts, co2_ts, eff, pbc,popN,
+        percepWindowSize=3,riskSens=1.0)
+        
+
+
+### TEST 3: Updating model n steps #####################################################
+
+if False: ### WORKS!
+    co2_ts=np.linspace(290, 300, 4)
+    popTotal=7130010000 # Wolfram: QuantityMagnitude[CountryData["World", "Population"]]
+    popN=sc.popIntoNgroups(popTotal,nGroups=10)
+    pcE_ts=sc.randomNormalF(5.049, 0.5, 10)
+    pcE_ts=np.atleast_2d(pcE_ts).transpose()
+    tData_ts=np.array([0,0.1,0.2,0.1]) # temperature
+    eff=sc.efficacyF(10)
+    pbc=sc.perceivedBehavioralControlF(10)
+    percepWindowSize=3
+    riskSens=1.0
+    
+    pcE_ts,tData_ts, co2_ts=sc.iterateNsteps(pcE_ts,tData_ts, co2_ts, 5, eff, pbc,popN,
+            percepWindowSize,riskSens)
+
+ 
+   
+     
+       
+         
+           
+               
+    # tData_ts.plot()
+    import matplotlib.pyplot as plt
+    hold(False)
+    
+    plt.plot(tData_ts)
+    plt.xlabel('Year')
+    plt.ylabel('T')
+    
+    plt.plot(co2_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+    
+    plt.plot(pcE_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+
+
+
+    
+    
+###### iterating function ################################################################
+
+if False:
+    
+    for i in range(10):
+        print i
+        pcE_ts,tData_ts,co2_ts = iterateOneStep(pcE_ts,tData_ts, co2_ts, eff, pbc,popN,
+            percepWindowSize=3,riskSens=1.0)
+        
+    # tData_ts.plot()
+    import matplotlib.pyplot as plt
+    hold(False)
+    
+    plt.plot(tData_ts)
+    plt.xlabel('Year')
+    plt.ylabel('T')
+    
+    plt.plot(co2_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+    
+    plt.plot(pcE_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+    
+    
+
+
+
+
+
+################ Testing of climate portion of model against observations #####
+if False:
+
+
+
+### Importing real data ################
+    
+    # CO2 data
+    co2Data=pd.read_excel("data/co2-atmospheric-mlo-monthly-scripps.xls", 
```

Comment:
Now that master was merged into your branch, you could use my new functions in here.